Assets/Scripts/InterceptorOrigins/InterceptorOrigin.cs

```diff
@@ -0,0 +1,136 @@
+  public string OriginId => _originConfig?.id ?? "Unknown";
+
+  // Override IsAssignable to indicate origins are not assignable as targets
+  public override bool IsAssignable() {
+    return false;
+  }
+
+  // Sets the origin configuration for this GameObject.
+  // Parameters:
+  //   config: The origin configuration
+  public void SetOriginConfig(InterceptorOriginConfig config) {
+    _originConfig = config;
+    _startPosition = transform.position;
+    _startTime = Time.time;
+  }
+
+  // Gets the origin configuration.
+  // Returns: The origin configuration
+  public InterceptorOriginConfig GetOriginConfig() {
+    return _originConfig;
+  }
+
+  // Gets the distance from this origin to a target position.
+  // Parameters:
+  //   targetPosition: Target position in world space
+  // Returns: Distance in meters
+  public float GetDistanceToTarget(Vector3 targetPosition) {
+    return Vector3.Distance(GetPosition(), targetPosition);
+  }
+
+  // Checks if this origin supports launching the specified interceptor type.
+  // Parameters:
+  //   interceptorType: Interceptor type to check
+  // Returns: True if supported
+  public bool SupportsInterceptorType(string interceptorType) {
+    return _originConfig?.SupportsInterceptorType(interceptorType) ?? false;
+  }
+
+  // Checks if this origin has available capacity.
+  // Returns: True if capacity available
+  public bool HasCapacity() {
+    return _originConfig?.HasCapacity() ?? false;
+  }
+
+  // Gets the available interceptor capacity.
+  // Returns: Number of interceptors that can still be allocated
+  public int GetAvailableCapacity() {
+    return _originConfig?.GetAvailableCapacity() ?? 0;
+  }
+
+  // Allocates an interceptor from this origin.
+  // Returns: True if allocation successful
+  public bool AllocateInterceptor() {
+    return _originConfig?.AllocateInterceptor() ?? false;
+  }
+
+  // Releases an interceptor back to this origin.
+  public void ReleaseInterceptor() {
+    _originConfig?.ReleaseInterceptor();
+  }
+
+  protected override void Start() {
+    base.Start();
+    _startPosition = transform.position;
+    _startTime = Time.time;
+  }
+
+  // Called when the origin is selected in the editor or for debugging.
+  protected virtual void OnDrawGizmosSelected() {
+    if (_originConfig != null) {
+      // Draw capacity indicator
+      Gizmos.color = HasCapacity() ? Color.green : Color.red;
+      Gizmos.DrawWireSphere(transform.position, 50f);
+
+      // Draw velocity vector for moving origins using actual rigidbody velocity
+      Vector3 actualVelocity = GetVelocity();
+
+      if (actualVelocity.magnitude > 0) {
+        Gizmos.color = Color.blue;
+        Vector3 velocityVector = actualVelocity.normalized * 200f;
+        Gizmos.DrawRay(transform.position, velocityVector);
+
+        // Draw movement trail
+        Gizmos.color = Color.cyan;
+        Vector3 startPos = _startPosition;
+        Vector3 currentPos = transform.position;
```

Comment:
I know it's pedantic, but prefer writing out words, so `startPosition` and `currentPosition` for better clarity